lib/utils/SOAPHeader.js

```diff
@@ -44,29 +44,30 @@ const PASSWORD_TEXT = 'http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-u
     NO_PASSWORD_TAG
   } = require('../security/constants/SecurityConstants');
 
-  class SOAPHeader {
+class SOAPHeader {
 
-    constructor(xmlParser){
-      this.parserAttributePlaceHolder = xmlParser.attributePlaceHolder;
-    }
+  constructor(xmlParser) {
+    this.parserAttributePlaceHolder = xmlParser.attributePlaceHolder;
+  }
 
   /**
    * Return the object for username token to parse
    * @param {UsernameTokenInput} usernameTokenInput token information
    * @param {object} jObj object to parse
+   * @param {string} parserAttributePlaceHolder character for attributes of parser
    * @returns {object} jObj object to parse
    */
-  processUsernameToken = (usernameTokenInput, jObj) => {
+  processUsernameToken = (usernameTokenInput, jObj, parserAttributePlaceHolder) => {
```

Comment:
@LuisTejedaS This seems to be the wrong definition of the function of class. it should be as following.

```
processUsernameToken (usernameTokenInput, jObj, parserAttributePlaceHolder) {
```

Reply:
Did this change failed any test? Ideally it should, same for below.

Reply:
@VShingala i would change those definitions, it does not fail any test because it is not an error, if we want to avoid this arrow definition inside a class (which is the best according to some reading i did) we could add it as a lint rule

Reply:
got it, please do make those changes. I meant the lint test here as this is not compatible with es2015 (it's an experimental feature for that version [source](https://stackoverflow.com/questions/31362292/how-to-use-arrow-functions-public-class-fields-as-class-methods) ). sorry for the confusion.

Reply:
we can add this rule to lint test as well

Reply:
yes the change is done, and @Carlos-Veloz is looking to add the rule